modulemd/v2/include/modulemd-2.0/private/modulemd-util.h

```diff
@@ -76,6 +76,9 @@ modulemd_hash_table_deep_str_set_copy (GHashTable *orig);
 GHashTable *
 modulemd_hash_table_deep_str_str_set_copy (GHashTable *orig);
 
+GEqualFunc
```

Comment:
```suggestion
gboolean
```

A GEqualFunc is actually a pointer type, which is why you're getting type errors. It describes a pointer to a function that must return `gboolean`.